src/ws/cockpitwebservice.c

```diff
@@ -1103,12 +1131,25 @@ on_session_closed (CockpitTransport *transport,
             {
               if (web_socket_connection_get_ready_state (socket->connection) == WEB_SOCKET_STATE_OPEN)
                 {
-                  payload = build_control ("command", "close",
-                                           "channel", channel,
-                                           "problem", problem,
-                                           "host-key", key,
-                                           "host-fingerprint", fp,
-                                           NULL);
+                  object = build_json ("command", "close",
+                                       "channel", channel,
+                                       "problem", public_problem,
+                                       "host-key", key,
+                                       "host-fingerprint", fp,
+                                       "host-key-failure", hostkey_failure,
```

Comment:
Where is this documented? Also what is the reasoning behind not just using varying problem codes?